superagi/helper/models_helper.py

```diff
@@ -0,0 +1,231 @@
+            return model  # Return error message if model not found
+
+        # Check the 'source_name' from ModelsConfig table
+        if not end_point and model["source_name"] not in ['OpenAI', 'Google Palm']:
+            return {"error": "End Point is empty or undefined"}
+
+        try:
+            model = Models(
+                model_name=model_name,
+                description=description,
+                end_point=end_point,
+                token_limit=token_limit,
+                model_provider_id=model_provider_id,
+                type=type,
+                org_id=self.organisation_id
+            )
+            self.session.add(model)
+            self.session.commit()
+
+        except Exception as e:
+            logging.error(f"Unexpected Error Occured: {e}")
+            return {"error": "Unexpected Error Occured"}
+
+        return {"success": "Model Details stored successfully"}
+
+    def fetchModels(self) -> List[Dict[str, Union[str, int]]]:
+        try:
+            models = self.session.query(Models.id, Models.model_name, Models.description, ModelsConfig.source_name).join(ModelsConfig, Models.model_provider_id == ModelsConfig.id).filter(Models.org_id == self.organisation_id).all()
+
+            result = []
+            for model in models:
+                result.append({
+                    "id": model[0],
+                    "name": model[1],
+                    "description": model[2],
+                    "model_provider": model[3]
+                })
+
+        except Exception as e:
+            logging.error(f"Unexpected Error Occured: {e}")
+            return {"error": "Unexpected Error Occured"}
+
+        return result
+
+    def fetchModelDetails(self, model_id: int) -> Dict[str, Union[str, int]]:
+        try:
+            model = self.session.query(
+                Models.id, Models.model_name, Models.description,
+                Models.end_point, Models.token_limit, Models.type,
+                ModelsConfig.source_name,
+            ).join(
+                ModelsConfig, Models.model_provider_id == ModelsConfig.id
+            ).filter(
+                and_(Models.org_id == self.organisation_id, Models.id == model_id)
+            ).first()
+
+            if model:
+                return {
+                    "id": model[0],
+                    "name": model[1],
+                    "description": model[2],
+                    "end_point": model[3],
+                    "token_limit": model[4],
+                    "type": model[5],
+                    "model_provider": model[6]
+                }
+            else:
+                return {"error": "Model with the given ID doesn't exist."}
+
+        except Exception as e:
+            logging.error(f"Unexpected Error Occured: {e}")
+            return {"error": "Unexpected Error Occured"}
+
+    def create_call_log(self, agent_execution_name: str, agent_id: int, tokens_consumed: int, tool_used: str, model: str) -> Optional[CallLogs]:
```

Comment:
move this to CallLogHelper in apm folder and shift this function.

Reply:
Done